Multisig/UI/Settings/Safe Settings/LoadableSafeSettingsViewModel.swift

```diff
@@ -48,24 +64,32 @@ class LoadableSafeSettingsViewModel: ObservableObject {
                 }
                 self.isLoading = false
             }, receiveValue: { response in
+                self.lastUpdatedAddress = safe.address
                 safe.update(from: response)
             })
             .store(in: &subscribers)
     }
     
 }
 
-extension Safe {
+extension LoadableSafeSettingsViewModel: NSFetchedResultsControllerDelegate {
+    func controllerDidChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
+        guard let safe = controller.fetchedObjects?.first as? Safe else { return }
+        if safe.address != lastUpdatedAddress {
+            self.safe = safe
+            updateOnce(safe)
+        }
+    }
+}
 
+extension Safe {
     func update(from safeInfo: SafeStatusRequest.Response) {
-        objectWillChange.send()
-        threshold = safeInfo.threshold
+        threshold = Int32(safeInfo.threshold)
         owners = safeInfo.owners
         masterCopy = safeInfo.masterCopy
         version = safeInfo.version
-        nonce = safeInfo.nonce
+        nonce = Int32(safeInfo.nonce)
```

Comment:
we really need to have the response validation, as the values can potentially overflow the Int32. We have this as a different tech debt ticket already.